sys/vm/vm_map.c

```diff
@@ -5246,6 +5247,13 @@ vmspace_fork(struct vmspace *vm1, vm_ooffset_t *fork_charge)
 			new_entry->cred = NULL;
 
 			vm_map_entry_link(new_map, new_entry);
+			/*
+			 * Install as rev_entry if required before adding
+			 * to the quarantine list so we don't get merged
+			 * with a newly quarantined neighbor.
+			 */
+			if (old_entry == old_map->rev_entry)
+				new_map->rev_entry = new_entry;
```

Comment:
vm_map_entry_quarantine() will compare the neighbours of `new_entry` with `new_map->rev_entry` to see if they can be merged with `new_entry`. So this assignment doesn't prevent merging from happening at the point where this entry is being added. It's not obvious to me that this is what we want, but I'm also not too familiar with the quarantine mechanism yet so could easily be missing something.

Reply:
There's a further check in `vm_map_mergeable_neighbors()` (called from `vm_map_try_merge_entries()`) that will prevent the entry in rev_entry from being merged. That suggests the checks in `vm_map_entry_quarantine()` are redundant.

As background, we can't allow the entry pointed to buy `rev_entry` to be merged (enlarged) because we may have already examined capabilities that were outside the current entry, but would be inside the new one and thus we'd fail to revoke those capabilities.

Reply:
Got it, thank you.